### DISCUSSION TITLE
Charts: counting rows of each option picker value

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1955] [Help] [answered]
Hey All,

Please make possible to use charts for showing how many rows of each option peaker value.
For example, I have a table with column "Type" - option picker - with three types: "A" , "B" , "C"
I'd like to see on graph how many records of each option (A, B and C) in the table stored.

Currently it's impossibe:

<img width="1790" alt="Screenshot 2021-07-08 at 23 04 10" src="https://user-images.githubusercontent.com/53024832/124984270-acb57000-e041-11eb-888a-dfccd425f20e.png">

Thanks.

**Comment by shogunpurple** [comment, marked as answer]
@sergey-dme what does your data look like? Is it an internal table?

If so, you should be able to use a view for this:

https://docs.budibase.com/data/views

Here's a simple example, assuming that you have some data with a "type"

### Step 1: Set up table
<img width="1650" alt="Screenshot 2021-07-09 at 11 15 15" src="https://user-images.githubusercontent.com/11256663/125063060-f1272700-e0a6-11eb-9e47-523e8ff40e93.png">

### Step 2: Create Count view
Next, you need to create a view. I've just noticed that there seems to be a bug when you use a view name with a space in it, so stay to names without spaces for now.

Set up a **count** view using the Auto ID - this will just count all the elements in the table.
<img width="665" alt="Screenshot 2021-07-09 at 11 16 28" src="https://user-images.githubusercontent.com/11256663/125063213-1ddb3e80-e0a7-11eb-9546-4d660fcfa351.png">

Now, group the counts by Type.
<img width="587" alt="Screenshot 2021-07-09 at 11 16 46" src="https://user-images.githubusercontent.com/11256663/125063257-292e6a00-e0a7-11eb-8734-640681bafb0c.png">

Here's the result:
<img width="1361" alt="Screenshot 2021-07-09 at 11 17 38" src="https://user-images.githubusercontent.com/11256663/125063348-46fbcf00-e0a7-11eb-813b-e6e144fc5113.png">


### Step 3: Show chart with statistics
You can now use your view in the design section. Use a data provider to bind to your view, then render the pie chart. Use `group` as the label column and `value` as the data column.

<img width="1644" alt="Screenshot 2021-07-09 at 11 19 46" src="https://user-images.githubusercontent.com/11256663/125063599-93470f00-e0a7-11eb-8e90-472b8d8927b3.png">

Hope this helps!

**Reply by sohailna** [reply]
Solved my problem too - just what I was searching for, thanks! 🥳 

Can this explanation  please be added to the documentation. Will save any future users from having to dig through the docs, then search github, then skim through two dozen discussions and issues to eventually find this one.

**Reply by phamalda** [reply]
Thank you so much for these detailed steps! I have been looking for this for so long! This got me where I needed to be! 👍

**Comment by sergey-dme** [comment]
@shogunpurple That's brilliant! Thank you very much! Views & Calculations are amazing & powerful, very surprised! 
But... I got it working on my second attempt: I already had 6-7 views (just with the filter) for my table, today when I created new one with calculations (followed your instruction) the result was empty. I deleted all existing views and created it again, now it works, mb it is a bug, still not sure..
